libsql-wal/src/segment/list.rs

```diff
@@ -337,6 +334,46 @@ impl<T> List<T> {
     }
 }
 
+mod send_fst_ops {
+    use std::ops::{Deref, DerefMut};
+    use std::sync::Arc;
+
+    use fst::map::{OpBuilder, Union};
+
+    /// Safety: Union contains a Box<dyn trait> that doesn't require Send, to it's not send.
+    /// That's an issue for us, but all the indexes we have are safe to send, so we're good.
```

Comment:
why are they safe? Can the compiler assume something here and create a ub? Would be good to expand this the logic here seems a bit weak.